xcdat/bounds.py

```diff
@@ -428,239 +467,238 @@ def _create_bounds(self, axis: CFAxisKey, coord_var: xr.DataArray) -> xr.DataArr
+        daily_freq: Optional[Literal[1, 2, 3, 4, 6, 8, 12, 24]] = None,
+    ) -> xr.DataArray:
+        """Creates time bounds for each timestep of the time coordinate axis.
 
-        if axis not in cf_axis_keys:
-            keys = ", ".join(f"'{key}'" for key in cf_axis_keys)
-            raise ValueError(
-                f"Incorrect 'axis' argument value. Supported values include {keys}."
-            )
+        This function uses ``_infer_freq`` to determine the temporal resolution and
+        then calls ``get_yearly_time_bounds``, ``get_monthly_time_bounds``, or
+        ``get_daily_time_bounds`` as appropriate. See the docstrings for these
+        functions for more details.
 
-        get_dim_coords(self._dataset, axis)
+        Parameters
+        ----------
+        time : xr.DataArray
+            The temporal coordinate variable for the axis.
+        freq : Optional[Literal["year", "month", "day", "hour"]]
+            The time frequency for creating time bounds, by default None (infer
+            the frequency).
+        daily_freq : Literal[1, 2, 3, 4, 6, 8, 12, 24], optional
+            If ``freq=="hour"``, this parameter sets the number of timepoints
+            per day for bounds, by default None. If greater than 1, sub-daily
+            bounds are created.
+
+            * ``daily_freq=None`` infers the freq from the time coords (default)
+            * ``daily_freq=1`` is daily
+            * ``daily_freq=2`` is twice daily
+            * ``daily_freq=4`` is 6-hourly
+            * ``daily_freq=8`` is 3-hourly
+            * ``daily_freq=12`` is 2-hourly
+            * ``daily_freq=24`` is hourly
 
+        Returns
+        -------
+        xr.DataArray
+            A DataArray storing bounds for the time axis.
 
-def create_yearly_time_bounds(time: xr.DataArray) -> xr.DataArray:
-    """Creates time bounds for each timestep with the start and end of the year.
+        Raises
+        ------
+        TypeError
+            If time coordinates are not composed of ``cftime.datetime`` object.
+        """
 
-    Bounds for each timestep correspond to Jan. 1 00:00:00 of the year of the
-    timestep and Jan. 1 00:00:00 of the subsequent year.
+        freq = _infer_freq(time) if freq is None else freq  # type: ignore
+
+        if freq == "year":
+            time_bnds = self._create_yearly_time_bounds(time)
+        elif freq == "month":
+            time_bnds = self._create_monthly_time_bounds(time)
+        elif freq == "day":
+            time_bnds = self._create_daily_time_bounds(time)
+        elif freq == "hour":
+            # Determine the daily frequency for generating time  bounds.
+            if daily_freq is None:
+                diff = time.values[1] - time.values[0]
+                hrs = diff.seconds / 3600
+                daily_freq = int(24 / hrs)  # type: ignore
+
+            time_bnds = self._create_daily_time_bounds(time, freq=daily_freq)  # type: ignore
```

Comment:
Do these functions support time coordinates composed of `datetime` objects? We should keep this support in this PR.

Reply:
I added logic in the latest commit to support datetime objects.